src/inc/utils/HashlistUtils.class.php

```diff
@@ -1147,6 +1147,15 @@ public static function getCrackedHashes($hashlistId, $user) {
       }
       $hashes[] = $arr;
     }
+    $entries_binary = Factory::getHashBinaryFactory()->filter([Factory::FILTER => [$qF1, $qF2]]);
```

Comment:
Instead of doing 2 queries, it is better to check in the hashlist whether it is a binary hashlist or a normal text hashlist. What you could do is the following:

if ($hashlist->getFormat === 0) {
    $entries = Factory::getHashFactory()->filter([Factory::FILTER => [$qF1, $qF2]]);
} else {
    $entries = Factory::getBinaryHashFactory()->filter([Factory::FILTER => [$qF1, $qF2]]);
}

And then you could build the hashes according to their type. In the HashlistUtils line 700 to line 714 you can see how hashes of different types are built, you can use that as example.

Reply:
Awesome! Thanks for pointing that out. I have implemented the recommended change